koku/masu/external/kafka_msg_handler.py

```diff
@@ -58,25 +58,26 @@
 from masu.prometheus_stats import KAFKA_CONNECTION_ERRORS_COUNTER
 from masu.util.ocp import common as utils
 
-LOG = logging.getLogger(__name__)
-
-EVENT_LOOP = asyncio.get_event_loop()
 
+LOG = logging.getLogger(__name__)
 HCCM_TOPIC = "platform.upload.hccm"
 VALIDATION_TOPIC = "platform.upload.validation"
 SUCCESS_CONFIRM_STATUS = "success"
 FAILURE_CONFIRM_STATUS = "failure"
+PRODUCER = Producer({"bootstrap.servers": Config.INSIGHTS_KAFKA_ADDRESS, "message.timeout.ms": 1000})
 
 
 class KafkaMsgHandlerError(Exception):
     """Kafka msg handler error."""
 
 
-def backoff(interval, maximum=64):  # pragma: no cover
-    """Exponential back-off."""
-    wait = min(maximum, (2 ** interval)) + random.random()
-    LOG.info("Sleeping for %.2f seconds.", wait)
-    time.sleep(wait)
+def delivery_callback(err, msg):
+    """Acknowledge message success or failure."""
+    if err is not None:
+        LOG.error(f"Failed to deliver message: {str(msg)}: {str(err)}")
+        raise KafkaMsgHandlerError(err)
```

Comment:
Does this callback happen in the same thread as `send_confirmation`?  normally callback mechanisms would not.. Just curious how this exception would be handled if that were the case

Reply:
Discussed, we addressed it